File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/VBNET/Source/Forms/GetErrorForm.vb
'----------------------------------------------------------------------------- 
' <copyright file="GetErrorForm.cs" company="KEYENCE">
'	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
' </copyright>
'----------------------------------------------------------------------------- 
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Forms
	Public Partial Class GetErrorForm
		Inherits Form
		#Region "Field"
		''' <summary>
		''' Maximum number of system errors to receive
		''' </summary>
		Private _rcvMax As Byte
		#End Region

		#Region "Property"

		''' <summary>
		''' Specify the maximum amount of system error information to receive.
		''' </summary>
		Public ReadOnly Property RcvMax() As Byte
			Get
				Return _rcvMax
			End Get
		End Property
		#End Region

		#Region "Event"
		''' <summary>
		''' Close start event
		''' </summary>
		''' <param name="e"></param>
		Protected Overrides Sub OnClosing(e As CancelEventArgs)
			If DialogResult = DialogResult.OK Then
				Try
					_rcvMax = Convert.ToByte(_txtboxErrCnt.Text)
				Catch ex As Exception
					MessageBox.Show(Me, ex.Message)
					e.Cancel = True
					Return
				End Try
			End If

			MyBase.OnClosing(e)
		End Sub
		#End Region

		#Region "Method"
		''' <summary>
		''' Constructor
		''' </summary>
		Public Sub New()
			InitializeComponent()

			' Field initialization
			_rcvMax = 0
		End Sub
		#End Region
	End Class
End Namespace

File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/CS/Source/Sample/Forms/GetErrorForm.cs
//----------------------------------------------------------------------------- 
// <copyright file="GetErrorForm.cs" company="KEYENCE">
//	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
// </copyright>
//----------------------------------------------------------------------------- 
using System;
using System.ComponentModel;
using System.Windows.Forms;

namespace LJV7_DllSampleAll.Forms
{
	public partial class GetErrorForm : Form
	{
		#region Field
		/// <summary>
		/// Maximum number of system errors to receive
		/// </summary>
		private byte _rcvMax;
		#endregion

		#region Property
		
		/// <summary>
		/// Specify the maximum amount of system error information to receive.
		/// </summary>
		public byte RcvMax
		{
			get { return _rcvMax; }
		}
		#endregion

		#region Event
		/// <summary>
		/// Close start event
		/// </summary>
		/// <param name="e"></param>
		protected override void OnClosing(CancelEventArgs e)
		{
			if (DialogResult == DialogResult.OK)
			{
				try
				{
					_rcvMax = Convert.ToByte(_txtboxErrCnt.Text);
				}
				catch (Exception ex)
				{
					MessageBox.Show(this, ex.Message);
					e.Cancel = true;
					return;
				}
			}

			base.OnClosing(e);
		}
		#endregion

		#region Method
		/// <summary>
		/// Constructor
		/// </summary>
		public GetErrorForm()
		{
			InitializeComponent();

			// Field initialization
			_rcvMax = 0;
		}
		#endregion
	}
}
